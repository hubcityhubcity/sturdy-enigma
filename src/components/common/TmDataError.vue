<template>
  <TmDataMsg icon="sentiment_very_dissatisfied" @on:intercom="handleIntercom()">
    <div slot="title">
      Aw shucks!
    </div>
    <div slot="subtitle">
      Even though you're connected a full node, we can't display this data for
      you right now. Please try again later or
      <a href="https://github.com/luniehq/lunie/issues">file a bug report</a> or
      <a class="intercom-button" href="#" @click="$emit('intercom')">
        open a support chat
      </a>
      . Apologies!
    </div>
  </TmDataMsg>
</template>

<script>
import TmDataMsg from "common/TmDataMsg"
import config from "src/../config"
import { mapState } from "vuex"
export default {
  name: `tm-data-error`,
  components: { TmDataMsg },
  computed: {
    ...mapState([`intercom`])
  },
  methods: {
    handleIntercom() {
      if (config.mobileApp) {
        this.$store.dispatch(`displayMessenger`)
      }
    }
  }
}
</script>
