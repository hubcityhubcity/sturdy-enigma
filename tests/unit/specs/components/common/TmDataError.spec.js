import { mount } from "@vue/test-utils"
import TmDataError from "common/TmDataError"

jest.mock(`src/../config.js`, () => ({
  mobileApp: true
}))

describe(`TmDataError`, () => {
  let wrapper
  beforeEach(() => {
    wrapper = mount(TmDataError)
  })

  it(`has the expected html structure`, () => {
    expect(wrapper.element).toMatchSnapshot()
  })

  it(`has an icon`, () => {
    expect(
      wrapper
        .find(`.tm-data-msg__icon i.material-icons`)
        .text()
        .trim()
    ).toBe(`sentiment_very_dissatisfied`)
  })

  it(`has a title`, () => {
    expect(
      wrapper
        .find(`.tm-data-msg__title div`)
        .text()
        .trim()
    ).toBe(`Aw shucks!`)
  })

  it(`has a subtitle`, () => {
    expect(
      wrapper
        .find(`.tm-data-msg__subtitle div`)
        .text()
        .trim()
    ).toContain(
      `Even though you're connected a full node, we can't display this data`
    )
  })

  it(`handleIntercom should dispatch displayMessenger action`, () => {
    const $store = { dispatch: jest.fn() }
    const self = {
      $store
    }
    TmDataError.methods.handleIntercom.call(self)
    expect($store.dispatch).toHaveBeenCalledWith(`displayMessenger`)
  })
})
