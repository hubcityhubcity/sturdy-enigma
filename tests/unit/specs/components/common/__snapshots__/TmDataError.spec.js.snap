// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TmDataError has the expected html structure 1`] = `
<div
  class="tm-data-msg"
>
  <div
    class="tm-data-msg__icon"
  >
    <i
      class="material-icons"
    >
      sentiment_very_dissatisfied
    </i>
  </div>
   
  <div
    class="tm-data-msg__text"
  >
    <h2
      class="tm-data-msg__title"
    >
      <div>
        
    Aw shucks!
  
      </div>
    </h2>
     
    <h3
      class="tm-data-msg__subtitle"
    >
      <div>
        
    Even though you're connected a full node, we can't display this data for
    you right now. Please try again later or
    
        <a
          href="https://github.com/luniehq/lunie/issues"
        >
          file a bug report
        </a>
         or
    
        <a
          class="intercom-button"
          href="#"
        >
          
      open a support chat
    
        </a>
        
    . Apologies!
  
      </div>
    </h3>
  </div>
</div>
`;
